apps/frontend/src/utils/auth.util.ts

```diff
@@ -9,11 +9,15 @@ import type { User } from 'next-auth';
 
 export async function requireAuthorizedUser(options?: {
   allowedRoles?: UserRole[];
+  currentRoute?: string;
 }): Promise<User> {
   const user = await getServerUser();
 
   if (!user) {
-    redirect('/sign-in', RedirectType.replace);
+    redirect(
+      `/sign-in?redirectUri=${encodeURIComponent(options?.currentRoute ?? '')}`,
```

Comment:
if there is no `currentRoute` value, there is no need to include `redirectUri` query param at all.